homeassistant/util/__init__.py

```diff
@@ -248,16 +248,23 @@ class Throttle(object):
     Pass keyword argument `no_throttle=True` to the wrapped method to make
     the call not throttled.
 
+    Initialize the decorator with the optional `immediate_throttle=True`
```

Comment:
What would be the use case for this?

Reply:
It's described in the referenced PR #14866

Reply:
That's a bad use case. Just don't have Home Assistant call `update` when it starts up by not passing `True` to `add_devices`.

Reply:
Well, the issue is not the initialisation. If I want to change the throttling interval *after* and *within* an update, simply executing
```
self.update = Throttle(newinterval)(self._update)
```
as I see it done (at init time only) in other sensors would not work: the `update` method is multithreaded and the above would be re-executed N times without any throttling.

Surely I don't know a good part of the Home Assistant code base, but I don't seem to find a way to adjust the throttling in that case. Of course any advice is more than welcome.

Reply:
If you need advanced throttling or rate limiting behavior, you'll need to create your own class.

Reply:
I see. In that case, would that new class live in the `util` package or what would be the recommended place for it?
To be noted that the issue at hand - minimizing the delay in getting fresh data whilst not hammering the remote cloud - is IMHO rather common across all the 'cloud polling' sensors. This is why I thought that enhancing the existing `Throttle` class would be beneficial.

Reply:
I still don't see how the existing Throttle can't help with that. The fact that it works for the rest of Home Assistant…

If you make your own, don't put it in util but keep it in the file that is using it.

Reply:
Well, how can I reset the `min_time` of the existing Throttle, without accessing the `_throttle` tuple (which I'd consider a violation of the encapsulation principle)?
All sensors in Home Assistant use a constant throttling interval. Perhaps I'm wrong but I'm saying that more than one sensor would benefit from an adaptive throttling interval.